app/uk/gov/hmrc/apiplatform/modules/submissions/domain/models/Submission.scala

```diff
@@ -67,15 +66,81 @@ object Submission {
     def random: Id = Id(UUID.randomUUID().toString())
   }
 
+  val create: (
+      String,
+      Submission.Id,
+      ApplicationId,
+      DateTime,
+      NonEmptyList[GroupOfQuestionnaires],
+      QuestionIdsOfInterest) => Submission = (requestedBy, id, applicationId, timestamp, groups, questionIdsOfInterest) => {
+
+      val initialStatus = Submission.Status.Created(timestamp, requestedBy)
+      val initialInstances = NonEmptyList.of(Submission.Instance(0, Map.empty, NonEmptyList.of(initialStatus)))
+    Submission(id, applicationId, timestamp, groups, questionIdsOfInterest, initialInstances)
+  }
+  
+  val addInstance: (Submission.AnswersToQuestions, Submission.Status) => Submission => Submission = (answers, status) => s => {
+    val newInstance = Submission.Instance(s.latestInstance.index+1, answers, NonEmptyList.of(status))
+    s.copy(instances = newInstance :: s.instances)
+  }
+  
+  val changeLatestInstance: (Submission.Instance => Submission.Instance) => Submission => Submission = delta => s => {
+    s.copy(instances = NonEmptyList(delta(s.instances.head), s.instances.tail))
+  }
+
+  val addStatusHistory: (Submission.Status) => Submission => Submission = newStatus => s => {
+    require(Submission.Status.isLegalTransition(s.status, newStatus))
+    changeLatestInstance(_.copy(statusHistory = newStatus :: s.latestInstance.statusHistory))(s)
+  }
+
+  val changeStatusHistory: (Submission.Status => Submission.Status) => Submission => Submission = delta => s => {
+    val inStatus = s.latestInstance.statusHistory.head
```

Comment:
ditto other reviews - should we check this makes for a legal transition?

Reply:
Fix from TPA